sdk/ml/azure-ai-ml/tests/dsl/unittests/test_controlflow_pipeline.py

```diff
@@ -25,6 +25,15 @@ class CustomizedObject:
     pass
 
 
+class TestDoWhilePipelineUT(TestControlFlowPipelineUT):
+    def test_infer_dynamic_input_type_from_mapping(self):
+        # Pass None to dynamic input in do-while loop body, and provide it in mapping for next iteration,
+        # which is a valid case in federated learning.
+        from test_configs.dsl_pipeline.dynamic_input_do_while.pipeline import pipeline_job
+
+        assert pipeline_job._customized_validate().passed
```

Comment:
Nit: more specific check here? validate loop_body's input type here.

Reply:
Nice catch, will add one

Reply:
Add assert operation to loop body's input type